src/circle.rs

```diff
@@ -147,6 +150,132 @@ impl Iterator for CirclePathIter {
     }
 }
 
+/// A segment of a circle.
+///
+/// If `inner_radius > 0`, then the shape will be a doughnut segment.
```

Comment:
What if the inner radius is greater than the outer radius?

Reply:
Good point - I've made it so that the shape is not dependent on the radii being in the correct order.